src/dir.targets

```diff
@@ -187,6 +203,36 @@
     <Error Condition="'$(TestRunExitCode)' != '0'" Text="One or more tests failed while running tests from '$(MSBuildProjectName)' please check log for details!" />
   </Target>
 
+  <Target Name="CheckTestCategories"
+          AfterTargets="Test"
+          Condition="'$(TestDisabled)'!='True'">
+
+    <Error Condition="'$(RunTestsWithCategories)' != '' And '$(RunTestsWithoutCategories)' != ''"
+           Text="Specifying both RunTestsWithCategories and RunTestsWithoutCategories is ambiguous." />
+
+    <ItemGroup Condition="'@(RunTestsWithCategoriesItems)'!=''">
+      <!-- Intersection of TestCategoriesItems and RunTestsWithCategoriesItems -->
+      <TestCategoriesToRun Include="@(TestCategoriesItems)" Condition="'@(RunTestsWithCategoriesItems)'=='%(Identity)'" />
+    </ItemGroup>
+    <ItemGroup Condition="'@(RunTestsWithoutCategoriesItems)'!=''">
+      <!-- Run all test categories except specified -->
+      <TestCategoriesToRun Include="@(TestCategoriesItems)" />
+      <TestCategoriesToRun Remove="@(RunTestsWithoutCategoriesItems)" />
```

Comment:
I didn't know about that :) I'll test if it works as expected and modify if works good.